dlgr/griduniverse/experiment.py

```diff
@@ -1597,3 +1600,14 @@ def _last_state_for_player(self, player_id):
         id_matches = [p for p in players if int(p['id']) == player_id]
         if id_matches:
             return id_matches[0]
+
+    def is_complete(self):
```

Comment:
It's not clear to me what the best mechanism for determining experiment completion is. Ideally, I think we would allow any `working` `Participants` to finish (which is the default behavior of the experiment server), but expire any outstanding HITs that don't have associated `Participant` entries.

It does not appear that we currently have any way to keep track of open HITs from an experiment that aren't yet associated with a `Participant`. As a result, we don't have a way to expire such HITs before shutting down the experiment. For now we just ensure that all the HITs we created initially were given `Participant` entries and finished the experiment in some sense. I think this should work for GU, but obviously isn't appropriate for other experiments.

Reply:
What's the difference between `num_recruits` and `max_participants`?